pkg/machines/scripts/create_machine.sh

```diff
@@ -26,8 +26,10 @@ spiceSupported(){
         *) QEMU=qemu-system-$ARCH;
     esac
 
-    printf '{"execute":"qmp_capabilities"}\n{"execute":"query-spice"}\n{"execute":"quit"}' | \
-        $QEMU --qmp stdio --nographic -nodefaults  | grep -q '"enabled":'
+    # if qemu-system-* is missing, SPICE is disabled by default
+    type -P $QEMU > /dev/null && \
+        printf '{"execute":"qmp_capabilities"}\n{"execute":"query-spice"}\n{"execute":"quit"}' | \
+            $QEMU --qmp stdio --nographic -nodefaults  | grep -q '"enabled":'
```

Comment:
One possible improvement for this whole script could be that you can use domain capabilities "virsh domcapabilities" to get the emulator path for default domain and you can specify some parameters to that command.  In future we need to expose the spice availability in domain capabilities.